### ISSUE TITLE
Surface Pro 10 Support

### ISSUE THREAD

**Issue by ZenerDeveloper** [linux-surface/linux-surface#1608]
I want to preface saying I've been using Surface-Linux for a few years now (I previously had a Surface Go, and it works great.) However, I needed something more powerful for school, so I purchased a Surface Pro 10.

The Surface Pro 10 is listed as supported on the official documentation. However, there is no listing in the feature matrix for this model and some features aren't working as expected (or at all), so I'm creating this issue to list what works and what doesn't.

Environment
* Hardware model: Surface Pro 10 for Business
* Kernel version: 6.9.9-surface-1
    * version 6.10.10-1 doesn't work see #1560
* Distribution: Ubuntu

What works
* Keyboard
* Touchpad
* Wi-Fi
* Bluetooth
* Speakers
* Sensors (ambient light)
* Battery Status
* Performance Modes
* Buttons
* Suspend (Has weird issues waking from sleep and sometimes locks up, but _technically_ works.

What Isn't Applicable
* LTE
* SD Card Reader

What Doesn't work
* Tablet Mode
* Touchscreen
* Pen

What is Untested
* Other Sensors
* S0ix3
* Cameras (likely don't work)
* Hibernate (Though it probably works)
* NFC (Not sure how to test)

Additionally, as this is an odd device meant for business, I would be willing to test anything that needs testing (features, patches, etc.)

**Comment by ZenerDeveloper** [comment]
After upgrading to 6.12.3-2, I can confirm a couple of changes:

 * Tablet mode, touchscreen, and auto rotate work.
 * The eraser on the Slim Pen 2 works, but not the tip.
 * Suspend and Shutdown entirely do not work. Both result in a blank screen and seemingly work, but the device will not wake up / boot without a force restart first. I also could not get Hibernate to work either.
 * The touchpad also lacks support for clicking and dragging with two separate fingers. You must click with one finger and drag with that same finger.

**Comment by Moareulf** [comment]
I've just tried a Surface 10 with 5G for business (intel Core Ultra 7) on Fedora 41 and my experience is as above except:

Keyboard not working
Wifi not working (Ath12k missing firmware ID - Failed to fetch board data for vendor 17cb, Device 1107)
5G device detected but not functioning
Suspend and shutdown not working

Have tried ubuntu as as check, same results on all items. Will submit the diagnostics in a new issue.

**Comment by ZenerDeveloper** [comment]
> [diagnostics.tar.gz](https://github.com/user-attachments/files/18272312/diagnostics.tar.gz) I've just tried a Surface 10 with 5G for business (intel Core Ultra 7) on Fedora 41 and my experience is as above except:
> 
> Keyboard not working Wifi not working (Ath12k missing firmware ID - Failed to fetch board data for vendor 17cb, Device 1107) 5G device detected but not functioning Suspend and shutdown not working
> 
> Have tried ubuntu as as check, same results on all items. Will submit the diagnostics here

Hmmm that's strange. I couldn't even get Fedora to install on my surface, it just gets stuck on a blank screen after selecting the grub menu option. I have the flex keyboard (one with Bluetooth as well as the connector) so I'm not sure if that would make a difference with the keyboard.

I didn't know there was a 5g variant, and after looking on the Microsoft website, it appears it supports Wifi 7 vs 6e, so the chips are probably different between the two models.

**Comment by fjkorol** [comment]
Hi @ZenerDeveloper! Could you please provide an updated review of the SP10? I currently have an SP10 in my cart and am expecting good results on Linux to complete. Thanks in advance!

**Comment by OwlInSpace** [comment]
> Hi [@ZenerDeveloper](https://github.com/ZenerDeveloper)! Could you please provide an updated review of the SP10? I currently have an SP10 in my cart and am expecting good results on Linux to complete. Thanks in advance!

Honestly, I wouldn't recommend it.  I had an SP8, and while linux-surface is absolutely awesome and the maintainers are doing their best here, the device itself overheated and frequently had issues [unrelated to Linux] like the keyboard deciding magically to stop working or the device deciding to turn itself on randomly while in a bag. It was a good device when it actually worked (the first few months of owning it) though, and the pen was great. In the end, one deft fall and the display backlight broke, and then I realized the device had absolutely no repairability and I basically had a paperweight. I ended up getting a ThinkPad, and I much prefer it because it just works and doesn't bother me like the SP8 did.

**Comment by ZenerDeveloper** [comment]
@fjkorol Honestly, while I have had no dealbreakers with the surface, it isn't ideal running Linux. Until things get better, I would recommend against buying one with the express purpose of installing Linux. I have even installed arch and tried the absolute latest kernel, with no success...

The issues in order of importance:
1. Shutdown DOES NOT WORK. The device will occasionally kernel panic, but usually the display goes black but the device continues running, requiring a force reboot. This is bad as a student because the device is in a bag 50% of the time.
2. Suspend is flakey. Arch with the latest kernel (6.12.7-1) seems to have fixed the issue, but I'm not sure if it was linux-surface or just the latest linux kernel. Regardless, as a student, I don't _love_ running arch on my main device, and the issues still exist with shutdown.
3. Surface Slim Pen 2 doesn't work quite right either. The touchscreen works as does the eraser, but the pen tip just does... nothing. I'm not sure why but it still exists on the latest kernel if I recall.
4. My flex keyboard only works when connected. I figured that would be the case, but if you are not detered by the other issues, do not spend the premium for the flex keyboard, just go for a regular one.

Having already bought the device, it runs linux very smoothly. I use gnome and the gestures and animations are all smooth (but enable 120hz which is off by default in both windows and linux). I can't complain too much, I think it's reasonable. Would I have bought it if I knew it would struggle this much? Nope.

TL;DR It's good at running linux but a terrible linux laptop.

**Comment by ocean1ee1** [comment]
The shutdown issue, I did not try to find the root issue, I'd like share one workaround that I daily used everyday.

When I want to shutdown, take the following steps:
1. use reboot
2. reboot to grub menu
3. use [shutdown] in grub menu [I already added a custom menu to grub menu]

**Comment by ZenerDeveloper** [comment]
@ocean1ee1 Yes I created a similar entry, but, as I'm tri booting Windows, Manjaro and Ubuntu, it's frustrating if I ever forget. I suspect it is related to the similar issue on the surface pro 9, but I don't know quite what the issue is there or what to check.

**Comment by ZenerDeveloper** [comment]
I would like to offer a correction for the Surface Slim Pen issues, after booting into Windows 11 to take notes for class... the pen tip also doesn't work under windows. I presume the pen tip will work under Linux if the actual hardware itself wasn't faulty. I am ordering a warranty replacement and I will report back if nobody else can test it first. (6-8 business days... sigh)

**Comment by ZenerDeveloper** [comment]
I got a new pen from best buy to hold me over until the replacement comes, and it turns out that both Ubuntu and Manjaro can utilize the pen. I will update the wiki entry in a moment.
For anyone curious, I swapped the pen nibs between the two and that did not solve the issue. The pen's internals have failed after only about 5 months of semi-infrequent use...

**Comment by Moareulf** [comment]
> I will update the wiki entry in a moment.

The entry as its listed at the moment I feel is misleading depending on the variant - we've got a lot of support ticked and its not accurate. The units with Wifi 7 and 5G are definitely not supported at all, no cover, wifi or other support. Perhaps it needs to be split by variant or have a new note there as if someone reads that at the moment they may be quite misled.

**Comment by ZenerDeveloper** [comment]
@Moareulf If you don't mind testing and editing the wiki [Here](https://github.com/linux-surface/linux-surface/wiki/Supported-Devices-and-Features/_edit), you can create a separate column for the 5G variant pretty easily (eg. SP10 (5G) ), and add your observations. If you add to the wiki, make sure you preview first, or, if you're not comfortable editing, you can post the results here and I can update it accordingly.

Ps. I find it very strange that the 5G variant has such limited support compared with the wifi variant. However, you're on Fedora, which I can't even get the installer to load. I would try eliminating Fedora as the issue before updating the wiki. Manjaro has support for the keyboard and mouse out of the box (no touchscreen, so it's not the full Surface kernel). I would try installing that and installing the Surface kernel and testing under that or potentially another distro.

After the support page is finished (for now) I'll go ahead and close this issue as it was created mainly to update the wiki.

**Comment by Moareulf** [comment]
Yeah, I've tried Ubuntu, Manjaro and Fedora same on all 3. No keyboard or trackpad at all.

I'll do that update on the wiki and raise a new issue for this.

**Comment by ZenerDeveloper** [comment]
Cool I'll go ahead and close this issue then.

**Comment by 7HE-W0R1D** [comment]
> [@fjkorol](https://github.com/fjkorol) Honestly, while I have had no dealbreakers with the surface, it isn't ideal running Linux. Until things get better, I would recommend against buying one with the express purpose of installing Linux. I have even installed arch and tried the absolute latest kernel, with no success...
> 
> The issues in order of importance:
> 
> 1. Shutdown DOES NOT WORK. The device will occasionally kernel panic, but usually the display goes black but the device continues running, requiring a force reboot. This is bad as a student because the device is in a bag 50% of the time.
> 2. Suspend is flakey. Arch with the latest kernel (6.12.7-1) seems to have fixed the issue, but I'm not sure if it was linux-surface or just the latest linux kernel. Regardless, as a student, I don't _love_ running arch on my main device, and the issues still exist with shutdown.
> 3. Surface Slim Pen 2 doesn't work quite right either. The touchscreen works as does the eraser, but the pen tip just does... nothing. I'm not sure why but it still exists on the latest kernel if I recall.
> 4. My flex keyboard only works when connected. I figured that would be the case, but if you are not detered by the other issues, do not spend the premium for the flex keyboard, just go for a regular one.
> 
> Having already bought the device, it runs linux very smoothly. I use gnome and the gestures and animations are all smooth (but enable 120hz which is off by default in both windows and linux). I can't complain too much, I think it's reasonable. Would I have bought it if I knew it would struggle this much? Nope.
> 
> TL;DR It's good at running linux but a terrible linux laptop.

Having the intel variant of the SP10 with Flex keyboard, I can confirm that with Fedora 42 and 6.14.2-1.surface.fc42.x86_64 kernel, point 3 and 4 here still hold true. `libinput debug-events` shows no event when one finger is pressed down and the other finger is moving. My touchscreen doesn't work at all and I didn't bother setting up the pen.

However, Shutdown is fixed with a kernel arg and I haven't had any problems with suspend.

Update: touchscreen works with the fedora official 6.14.9 kernel, but not the surface linux 6.14.2 kernel

**Comment by 7HE-W0R1D** [comment]
Click and drag is working now! Thanks to [our friends from framework](https://community.frame.work/t/fixed-trackpad-in-linux/3747/6). I applied the same quirk `AttrEventCode=-BTN_RIGHT`, rebooted, and its working! I'll create a PR for libinput tomorrow.

Update: PR merged to libinput
Update 2: Working OOTB with Fedora 42 now

**Comment by 7HE-W0R1D** [comment]
Touchscreen stopped working after suspend, and I fixed that with the follow module reload: 
```
sudo modprobe -r intel_quickspi intel_thc
sudo modprobe intel_quickspi intel_thc
```
(I'm using fedora official 6.14.11 kernel while fixing this).

Alternatively, the following code generated by AI turns that into a service so that's automatic, tested and working:

```
sudo tee /usr/local/bin/auto-fix-surface-touch.sh > /dev/null << 'EOF'
#!/bin/bash
echo "Auto-fixing Surface touchscreen after resume..."

modprobe -r intel_quickspi intel_thc
sleep 1
modprobe intel_quickspi intel_thc
sleep 1

echo "Surface touchscreen drivers reloaded successfully."
EOF

sudo chmod +x /usr/local/bin/auto-fix-surface-touch.sh
```

```
sudo tee /etc/systemd/system/surface-touch-resume.service > /dev/null << 'EOF'
[Unit]
Description=Fix Surface touchscreen after resume
After=suspend.target hibernate.target hybrid-sleep.target
Before=display-manager.service

[Service]
Type=oneshot
ExecStart=/usr/local/bin/auto-fix-surface-touch.sh
RemainAfterExit=yes
TimeoutSec=30

[Install]
WantedBy=suspend.target hibernate.target hybrid-sleep.target
EOF
```

```
sudo systemctl daemon-reload
sudo systemctl enable surface-touch-resume.service
```

Hopefully this helps someone!

**Comment by ZahidShah786** [comment]
I just installed arch with gnome to Surface pro 10 business (intel) and it's amazing. My first expression was touch screen works. It was broken on Ubuntu. Other thing will I test, but so far it's beautiful. I will never go back to Windows.
